src/lib.rs

```diff
@@ -355,8 +388,24 @@ pub fn run_tests<D: 'static + Send + Sync>(
     let mut num_benches = 0;
     let mut num_passed = 0;
 
+    let run_test = std::sync::Arc::new(run_test);
+    let run_test = move |test: &Test<D>| {
+        let run_test = run_test.clone();
+        std::panic::catch_unwind(std::panic::AssertUnwindSafe(move || {
+            run_test(test)
+        }))
+        .unwrap_or_else(|_e| {
+            Outcome::Failed { msg: None }
+        })
```

Comment:
Two things: the `clone` is not necessary and rustfmt again makes me unhappy.

```suggestion
        let run = std::panic::AssertUnwindSafe(|| run_test(test));
        std::panic::catch_unwind(run).unwrap_or_else(|_e| {
            Outcome::Failed { msg: None }
        })
```